### PR TITLE
[ARTS-212] Assign Role to a Person Automation API tests

### PR DESCRIPTION
**Pull request by sameerapurini**
## Description
The PR consists of api automation tests to check the positive response when a user assigns a role to a person in a particular site. The role has permissions added too using permissions endpoint. The data is pulled dynamically from the data js file added within the tests folder. 

I have also moved the global utilities file 'utils' to the root of api-tests hence had to update the path of it in role service and permission files.

And also as the site name and alias fields doesn't accept numbers but only alphabets I had to update the 'getRandomString' function in the 'utils' accordingly.

### Changes
- [ARTS-212] - api tests added to cover assign role endpoint
- [ARTS-595] - Test approach

### Checklist:

- [ ] Branch name follows convention (feature/arts-#-short-name OR fix/arts-#-short-name)
- [ ] I have included a short-meaningful title, description and changes for this PR

[ARTS-212]: https://ndph-arts.atlassian.net/browse/ARTS-212

[ARTS-595]: https://ndph-arts.atlassian.net/browse/ARTS-595

### CODE REVIEW

File: api-tests/data/assign-roles/assign-roles.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+const createPerson = {
+    "prefix": "Ms",
+    "givenName": "Cookie",
+    "familyName": "Cookie"
+}
+
+const createSite = {
+    "name": utils.getRandomString(5),
+    "alias": utils.getRandomString(3),
+    "parentSiteId": "5d01f252-d41c-40f8-b3c1-4bd7282fb28e",
```
Comment (alexbaxter) [inline api-tests/data/assign-roles/assign-roles.js:24]: Where are you getting this ID from? Should be dynamically generated at init time?

File: api-tests/specs/assign-roles/assign-roles.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+const requests = require('../../data/assign-roles/assign-roles')
+const conf = require('../../config/conf')
+const sitesEndpointUri = ':8083/sites';
```
Comment (alexbaxter) [inline api-tests/specs/assign-roles/assign-roles.js:3]: Can you get these service URLs from the config server? These are the local configurations I think, won't work in Azure maybe?

**Review by alexbaxter — COMMENTED** [review]
No changes but a couple of questions.

File: api-tests/data/assign-roles/assign-roles.js (hunk 1 above)
Comment (sameerapurini) [reply api-tests/data/assign-roles/assign-roles.js:24]: that's right @alexbaxter I was about to come to you with regards to this. I am not sure if the 'init-services' will be run in the pr pipeline for me to pick it up from there. Or else I can just add another line of code to handle this.

Comment (alexbaxter) [reply api-tests/data/assign-roles/assign-roles.js:24]: If you can reasonable assume the root site has been created, then do a GET on /sites and parse the ID out of that.

File: api-tests/specs/assign-roles/assign-roles.js (hunk 2 above)
Comment (sameerapurini) [reply api-tests/specs/assign-roles/assign-roles.js:3]: I might need help with this @alexbaxter

**Review by alexbaxter — APPROVED** [review]
Approved, questions answered.

File: api-tests/specs/assign-roles/assign-roles.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+const permissionsEndpointUri = ':8082/roles';
```
Comment (tamirkamara) [inline api-tests/specs/assign-roles/assign-roles.js:4]: Can you rename this to be "roles" endpoint?
Permissions are something different and might be confusing here.

**Review by tamirkamara — CHANGES_REQUESTED** [review]

**Review by alexbaxter — DISMISSED** [review]
Approved

**Review by tamirkamara — APPROVED** [review]

**Review by alexbaxter — APPROVED** [review]
Approved

**Pull request merged**